cmd/billing/billing.go

```diff
@@ -0,0 +1,225 @@
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+var usageCmd = &cobra.Command{
+	Use:   "usage",
+	Short: "View billing usage data available for the account in YugabyteDB Managed",
+	Long:  "View billing usage data available for the account in YugabyteDB Managed",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+		startDate, _ := cmd.Flags().GetString("start")
+		endDate, _ := cmd.Flags().GetString("end")
+		clusters, _ := cmd.Flags().GetStringSlice("clusters")
+		format, _ := cmd.Flags().GetString("format")
+		output, _ := cmd.Flags().GetString("output")
+		if output == "" {
+			output = "usage"
+		}
+
+		startDateTime, err := parseAndFormatDate(startDate)
+		if err != nil {
+			logrus.Fatalf("Invalid start date format. Use either RFC3339 format (e.g., '2023-09-01T12:30:45.000Z') or 'yyyy-MM-dd' format (e.g., '2023-09-01').\n")
+		}
+
+		endDateTime, err := parseAndFormatDate(endDate)
+		if err != nil {
+			logrus.Fatalf("Invalid end date format. Use either RFC3339 format (e.g., '2023-09-01T12:30:45.000Z') or 'yyyy-MM-dd' format (e.g., '2023-09-01').\n")
+		}
+
+		if startDateTime > endDateTime {
+			logrus.Fatalf("Start date must be before end date.")
+		}
+
+		startDate = startDateTime
+		endDate = endDateTime
+
+		respC, r, err := authApi.ListClustersByDateRange(startDate, endDate).Execute()
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		clusterData := respC.Data.GetClusters()
+		var clusterUUIDMap = make(map[string]string)
+
+		for _, cluster := range clusterData {
+			clusterName := cluster.Name
+			clusterUUIDMap[clusterName] = cluster.Id
+		}
+
+		var selectedUUIDs []string
+		if len(clusters) > 0 {
+			selectedUUIDs = make([]string, 0, len(clusters))
+			for _, clusterName := range clusters {
+				if clusterID, ok := clusterUUIDMap[clusterName]; ok {
+					selectedUUIDs = append(selectedUUIDs, clusterID)
+				} else {
+					logrus.Fatalf("Cluster name '%s' not found within the specified time range.\n", clusterName)
+				}
+			}
+		} else {
+			selectedUUIDs = nil
+		}
+
+		resp, r, err := authApi.GetBillingUsage(startDate, endDate, selectedUUIDs).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		usageData := resp.GetData()
+
+		switch strings.ToLower(format) {
+		case "csv":
+			if err := outputCSV(usageData, output); err != nil {
+				return
+			}
+		case "json":
+			if err := outputJSON(usageData, output); err != nil {
+				return
+			}
+		default:
+			logrus.Warnf("Unsupported format: %s. Defaulting to CSV.\n", format)
+			if err := outputCSV(usageData, output); err != nil {
+				return
+			}
+		}
+	},
+}
+
+func parseAndFormatDate(dateStr string) (string, error) {
```

Comment:
Why not return a DateTime type? Why convert it to string?

Reply:
Done

Reply:
Done